tests/http-adapter.spec.js

```diff
@@ -98,4 +96,103 @@ describe('HTTP adapter', function () {
       return adapter.destroy('/api/user/42/')
     })
   })
+
+  describe('#serializeRequests', function () {
+    it('parallel requests do not start when `serializeRequests = true`', async function () {
+      server.autoRespond = false
+      server.respondImmediately = false
+
+      // Turn on `serializeRequests` and then make two parallel requests
+      adapter.serializeRequests = true
+      adapter.get('/api/user/42/')
+      adapter.get('/api/offer/9000/')
+
+      // Tick microtask queue
+      await Promise.resolve().then(() => {})
+
+      expect(server.requests).toHaveLength(1)
+      expect(server.requests[0].url).toEqual('/api/user/42/')
```

Comment:
Maybe finish the queue to avoid pollution?

Reply:
You mean between the tests? Since `autoRespond` is off, any pending promises should never resolve and be garbage collected once `server` is.